Userland/Libraries/LibGfx/Point.h

```diff
@@ -42,6 +42,14 @@ class Point {
     {
     }
 
+    static constexpr Point from_polar(T radius, T angle) requires(IsFloatingPoint<T>)
```

Comment:
Why does that requires not do anything for Clang? It still tries to compile the constructor for IntPoint which expectedly fails.

Reply:
Clang might just be behind the C++20 game here.

Before C++20, this was a common issue with SFINAE - you couldn't do something like `std::enable_if<std::is_floating_point_v<T>>` on a method if `T` was a template on the containing class. So it was common to do something like:

```c++
template <typename U = T>
static constexpr std::enable_if_t<std::is_floating_point_v<U>, Point> from_polar(U radius, U angle)
{
}
```

...in order to make SFINAE work on the template types of that specific method.

I think what you've done is valid in C++20, but to make clang happy until it catches up, this should work:

```c++
template <typename U = T>
static constexpr Point from_polar(U radius, U angle) requires(IsFloatingPoint<U>)
{
}
```

Reply:
The function is not needed anyways, my local branch has it already deleted. So thanks for the info but it's not a problem anymore :)